src/tools/Line.ts

```diff
@@ -0,0 +1,159 @@
+import { Tool } from '@/tools/Tool'
+import Konva from 'konva'
+import { CanvasElement } from '@/types/Canvas'
+import uuid from 'uuid'
+import throttle from 'lodash.throttle'
+
+export default class Line implements Tool {
+  private line: Konva.Line
+  private arrow: Konva.Arrow
+  private TBar: Konva.Line
+  private mapRatio: number
+  private stroke: number[][]
+  constructor (public readonly name: string,
+               public size: number,
+               public colour: string,
+               public endStyle: string,
+               public strokeStyle: number,
+               public temporarily: boolean) {
+    this.line = new Konva.Line()
+    this.arrow = new Konva.Arrow()
+    this.TBar = new Konva.Line()
+    this.mapRatio = 1
+    this.stroke = [
+      [0, 0],
+      [30, 10]
+    ]
+  }
+
+  // eslint-disable-next-line
+  mouseDownAction = (e: Konva.KonvaPointerEvent, canvasElement: CanvasElement, layer: Konva.Layer, _socket: WebSocket): void => {
+    canvasElement.data = [e.evt.x, e.evt.y]
+    canvasElement.id = uuid()
+    canvasElement.endStyle = this.endStyle
+    canvasElement.strokeStyle = this.strokeStyle
+    this.line = new Konva.Line()
+    this.arrow = new Konva.Arrow()
+    this.TBar = new Konva.Line()
+    if (this.endStyle === 'line') {
```

Comment:
could you do something clever with `this[variable] = this.create[variable.toUpperCase()](canvasElement)`?
It will streamline it nicely :)

Reply:
You can get rid of all the if else statements then